[FEATURE] Affichage du changement de langue sur pix.org (PIX-1601).

MelanieMEB opened the pull request:
## :unicorn: Problème
Le seul moyen actuellement de changer la langue est l'URL. Il faut proposer un switcher aux utilisateurs, disponible uniquement sur .org

## :robot: Solution
- Ajout d'un menu dropdown pour le changement de langue, et d'un texte simple dans le burger menu

## :rainbow: Remarques
- Suppression de l'ancien bandeau gris de changement de langue
- Ajout d'un nouveau composant
- Je n'ai pas utilisé `this.$i18n.setLocale` car lorsqu'on changeait la locale avec, la page était tout de suite rechargée avec la même url. Si on était sur `/fr/les-tests`, il redirigeait vers `/en-gb/les-tests` qui n'existe pas (c'est `/en-gb/the-tests`)

## :100: Pour tester
Sur https://site-pr239.review.pix.fr :  ne pas voir le language switcher
Sur https://site-pr239.review.pix.org/fr : voir le language switcher
- Pouvoir changer la langue
- Pouvoir changer la langue quelque soit la page sur laquelle on est
- Pouvoir changer la langue au clavier
- Pouvoir changer la langue dans le burger menu

MelanieMEB committed:
BugFix : current dropdown does not work

pix-service commented:
I'm deploying this PR to these urls:

- Pix Site (fr): https://site-pr239.review.pix.fr
- Pix Site (org): https://site-pr239.review.pix.org
- Pix Pro (fr): https://pro-pr239.review.pix.fr
- Pix Pro (org): https://pro-pr239.review.pix.org

 Please check it out!

MelanieMEB committed:
Add a Component LanguageSwitcher

This component is a copy-past of previous Language Dropdown and Dropdown-button

MelanieMEB committed:
Remove old menu for language dropdown

MelanieMEB committed:
Change design of language switcher

MelanieMEB committed:
Improve accessibility

MelanieMEB committed:
Add a simple language switcher in burger menu

MelanieMEB committed:
Add correct svg

MelanieMEB commented on static/images/globe-blue.svg:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Utiliser SVGO

MelanieMEB commented on lang/fr.js:
```diff
@@ -1,6 +1,6 @@
 export default {
-  fr: 'Francophone',
-  'fr-fr': 'Français',
+  fr: 'Français',
+  'fr-fr': 'Franco-Français',
   'en-gb': 'English',
```
Version Française ?

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+    },
+  },
+  data() {
+    const allLocales = this.$i18n.locales.map((locale) => {
+      return { name: this.$t(locale.code), lang: locale.code }
+    })
+    const availableLocales = allLocales.filter(
+      (locale) => locale.lang !== 'fr-fr'
+    )
```
```suggestion
    const availableLocales = this.$i18n.locales.map((locale) => {
      return { name: this.$t(locale.code), lang: locale.code }
    })
    const languageLocales = availableLocales.filter(
      (locale) => locale.lang !== 'fr-fr'
    )
```

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+
+    hideMenu(event) {
+      if (event && event.keyCode) {
+        if (event.keyCode === 27) {
```
ESCAPE_KEYCODE = 27

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+          v-if="option !== currentLanguage"
+          class="language-switcher-text__lang"
+        >
+          <a href="javascript:void(0)" @click="update(option)">
+            {{ option.name }}
+          </a>
```
a tester 
```suggestion
          <a href="/ option.lang" @click="update(option)">
            {{ option.name }}
          </a>
```

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+}
+.language-switcher__button:hover,
+.language-switcher__button:focus {
+  background-image: url('/images/globe-blue.svg');
```
Voir fontawesome

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+}
+</script>
+
+<style lang="scss">
```
Problème d'alignement entre la planete (à rétrécir) et le separator puis les boutons

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+  height: 2rem;
+  background-size: 2rem;
+
+  padding: 4px 3px 0px 3px;
```
```suggestion
  padding: 4px 3px 0 3px;
```

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+<style lang="scss">
+.language-switcher {
+  display: flex;
+  fill: blue;
```
```suggestion
```

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+}
+.language-switcher__button {
+  background: none;
+  color: inherit;
```
```suggestion
```

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+    height: 1.5rem;
+  }
+}
+.language-switcher__button {
```
Passer en mode sass

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+  font-size: 14px;
+  text-align: center;
+  background-color: #fff;
+  border: 1px solid #ccc;
```
```suggestion
  border: 1px solid $grey-??;
```

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+.language-switcher__dropdown-menu > li > a {
+  padding: 0;
+  display: block;
+  clear: both;
```
```suggestion
```

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+  clear: both;
+  font-weight: normal;
+  line-height: 1.6;
+  color: #333333;
```
```suggestion
  color: $grey-??;
```

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+  display: block;
+  clear: both;
+  font-weight: normal;
+  line-height: 1.6;
```
```suggestion
  line-height: 1.6rem;
```

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+  padding: 0;
+  display: block;
+  clear: both;
+  font-weight: normal;
```
```suggestion
```

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+  text-decoration: none;
+}
+.language-switcher__dropdown-menu > li > a:hover {
+  background: #efefef;
+  color: #409fcb;
```
Voir les couleurs

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+  list-style: none;
+}
+
+ul li::before {
+  content: '';
+  margin: 0;
+}
```
```suggestion
ul li::before {
```

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+    const page = document.getElementsByTagName('body')[0]
+    const menu = document.getElementsByClassName('language-switcher')[0]
+    page.removeEventListener('click', this.hideMenu)
+    menu.addEventListener('keydown', this.hideMenu)
```
```suggestion
    menu.removeEventListener('keydown', this.hideMenu)
```

MelanieMEB commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,225 @@
+    },
+  },
+  mounted() {
+    const page = document.getElementsByTagName('body')[0]
+    const menu = document.getElementsByClassName('language-switcher')[0]
+    page.addEventListener('click', this.hideMenu)
+    menu.addEventListener('keydown', this.hideMenu)
```
```suggestion
    const page = document.getElementsByTagName('body')[0]
    page.addEventListener('click', this.hideMenu)
    const menu = document.getElementsByClassName('language-switcher')[0]
    menu.addEventListener('keydown', this.hideMenu)
```

Et sortir 2 méthodes : listenEscape et listenClickOutside
Voir pour utiliser this.$el

MelanieMEB replied:
Utiliser `@keydown.esc.exact ` ?

MelanieMEB committed:
Pass svgo on svg

MelanieMEB committed:
Cleanup CSS and simplify js interaction

sbedeau commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,198 @@
+    display: block;
+    height: 1.5rem;
+  }
+  &__button {
```
**suggestion(func):** ajouter `cursor: pointer`

sbedeau commented on components/LanguageSwitcher.vue:
```diff
@@ -0,0 +1,198 @@
+    float: left;
+    min-width: 100px;
+    padding: 5px 0;
+    margin: 2px 0 0;
+    margin-top: 35px;
+    margin-left: -30px;
```
suggestion: Je pense qu'on peut les regrouper ou alors juste enlever le `margin: 2px 0 0`
